src/main/java/com/develhope/spring/purchase_order/controller/PurchaseOrderController.java

```diff
@@ -0,0 +1,60 @@
+package com.develhope.spring.purchase_order.controller;
+
+import com.develhope.spring.purchase_order.dto.PurchaseOrderCreationDTO;
+import com.develhope.spring.purchase_order.dto.PurchaseOrderResponseDTO;
+import com.develhope.spring.purchase_order.entity.PurchaseOrder;
+import com.develhope.spring.purchase_order.entity.PurchaseOrderStatus;
+import com.develhope.spring.purchase_order.service.PurchaseOrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/purchase_order")
+public class PurchaseOrderController {
+    @Autowired
+    private PurchaseOrderService purchaseService;
+
+    @PostMapping
+    public ResponseEntity<PurchaseOrderResponseDTO> create(@RequestBody PurchaseOrderCreationDTO purchaseOrderCreationDTO) {
+        return ResponseEntity.ok(purchaseService.createOrder(purchaseOrderCreationDTO));
+    }
+
+    @GetMapping("/{id}")
+    public PurchaseOrder findById(@PathVariable long id) {
+        return purchaseService.findOrderById(id);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<PurchaseOrder>> findAll() {
+        return ResponseEntity.status(HttpStatus.FOUND).body(purchaseService.findAllOrders());
```

Comment:
diceva Lorenzo che il found si usa solo se c'è un redirect, altrimenti lo stato dovrebbe essere sempre ok

Reply:
non ti preoccupare dobbiamo comunque chiamare il ritorno delle chiamate in un dto, lo sto facendo in un altra pull request se accettate questa